pkg/integrations/v4/dm/emitter.go

```diff
@@ -149,16 +152,28 @@ func (e *emitter) runFwReqConsumer(ctx context.Context) {
 
 		case req := <-e.reqsQueue:
 			e.measure(instrumentation.Counter, instrumentation.DMDatasetsReceived, int64(len(req.Data.DataSets)))
+
+		loop:
 			for _, ds := range req.Data.DataSets {
 				select {
 				case _ = <-ctx.Done():
 					return
 				default:
 					if ds.IgnoreEntity {
-						// Should not contain entity attributes
-						e.processDatasetNoRegister(req.Data.Integration, req.FwRequestMeta, ds)
-					} else {
+						e.emidDatasetWithEmptyEntity(req.Data.Integration, req.FwRequestMeta, ds)
+						continue loop //nolint:nlreturn
+					}
+					if ds.Entity.IsAgent() {
+						e.emitDatasetForAgent(ctx, req.Data.Integration, req.FwRequestMeta, ds)
+						continue loop //nolint:nlreturn
+					}
+					deprecateRegisterFFEnabled, deprecateRegisterFFExists := e.ffRetriever.GetFeatureFlag(fflag.FlagDmRegisterDeprecated)
+					if !deprecateRegisterFFExists || (deprecateRegisterFFExists && !deprecateRegisterFFEnabled) {
```

Comment:
WDYT if we wrap this in a small helper function with a good name, that will improve the readability of this boolean expression?

Reply:
👍

Reply:
Changed in https://github.com/newrelic/infrastructure-agent/pull/1357/commits/06818844ac2e45012a39bcb66a82884b33c5ba19

wdyt?